os/fs/smartfs/smartfs_smart.c

```diff
@@ -651,6 +653,10 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 
 	header = (struct smartfs_chain_header_s *)fs->fs_rwbuffer;
 	byteswritten = 0;
+
+	/* As long as the current file pointer positions for writing is less than the total data length in the file, we are doing and overwrite operation
+	 * When the file pointer positins surpasses the file's data length, it becomes an append operation which will be handled by a separate function
```

Comment:
/* As long as the current file pointer "position" for writing is less than the total data length in the file, we are doing and "an"overwrite operation
	 * When the file pointer "position" surpasses the file's data length, it becomes an append operation which will be handled by a separate function